fe/src/main/java/org/apache/doris/analysis/StatementBase.java

```diff
@@ -124,6 +124,9 @@ public void castResultExprs(List<Type> types) throws AnalysisException {
         Preconditions.checkNotNull(resultExprs);
         Preconditions.checkState(resultExprs.size() == types.size());
         for (int i = 0; i < types.size(); ++i) {
+            if (resultExprs.get(i).getType().isDateType() && types.get(i).isDateType()) {
```

Comment:
Please add a detailed comment here to explain the problem. Specific examples can be used.

Reply:
OK